lib/devfs/include/devfs/device.h

```diff
@@ -190,22 +219,151 @@ devtoname(struct device *dev)
 	return dev->name;
 }
 
+/*
+ * Open the device specified by name
+ * Can be used to check whether the device exists or not
+ * 
+ * @param name
+ * The name of the device
+ * @param mode
+ * The open mode(should be handled by the device driver if needed)
+ * @param devp
+ * Return parameter. The opened device will be reachable in devp after
+ * the function returns
+ * @return
+ * Returns the result of the device's open routine
+ */
+
 int device_open(const char *name, int mode, struct device **devp);
+
+/*
+ * Close the device specified by dev
+ * Does not return error if the target driver does not have
+ * a close routine
+ * 
+ * @param dev
+ * The device to be closed
+ * @return
+ * Returns the result of the device's close routine
+ */
+
 int device_close(struct device *dev);
+
+/*
+ * Perform a scatter/gather read on the device pointed to by dev
+ * @param dev
+ * The device to read from
+ * @param uio 
+ * The structure containing the scatter-gather list
+ * @param ioflags
+ * The flags to be sent to the read operation
+ * @return
+ * The result of the read operation 
+ */
+
 int device_read(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Perform a scatter/gather write on the device pointed to by dev
+ * @param dev
+ * The device to write to
+ * @param uio
+ * The structure containing the scatter-gather list
+ * @param ioflags
+ * The flags to be sent to the write operation
+ * @return
+ * The result of the write operation
+ */
+
 int device_write(struct device *dev, struct uio *uio, int ioflags);
+
+/*
+ * Perform an I/O control request
+ * @param dev
+ * The device to perform the request on
+ * @param cmd
+ * The device dependent command to be executed
+ * @param arg
+ * The user buffer sent as argument to the command. The buffer
+ * must be validated by the ioctl routine of the driver
+ * @return
+ * The result of the I/O control request
+ */
 int device_ioctl(struct device *dev, unsigned long cmd, void *arg);
+
+/*
+ * Returns information about a device
+ * The target of the query should be placed in the cookie field
+ * of the parameter
+ * @param info
+ * Used as a return parameter. The device must be specified
+ * in the cookie field and on return the id, flags and name
+ * fields will be filled with the requested information.
+ * @return
+ * Returns 0 on success and ESRCH otherwise
```

Comment:
```suggestion
 *     Used as a return parameter. The device must be specified
 *     in the cookie field and on return the id, flags and name
 *     fields will be filled with the requested information.
 * @return
 *     Returns 0 on success and ESRCH otherwise
```

@calex257 , check also the discussion [here](https://github.com/unikraft/unikraft/pull/708/files)